Add live display of Treadmill Inclination

DXCanas opened the pull request:
This PR implements live display support for FTMS Treadmill Inclination. It relied heavily on various AI's/prompts, as this is not my typical stack nor my typical platform. But I am excited about this field, and so wanted to try my hand.

The last thing I want to do is add to the ever-growing mass of AI Slop. But this PR description, too, will be AI generated as I have only minimal idea what's being done here :)

Design Philosophy: I modeled this implementation directly after the existing Resistance Level feature.

My primary goal was to respect the project's architectural constraint of avoiding database bloat for sparse data. As discussed in Issue ##577 , adding a permanent database column for "Inclination" is undesirable.

Consequently, this feature is strictly Display Only:

The data is not persisted to the Isar database.

The data is not exported to CSV/FIT.

The feature is opt-in via a new User Preference.

Implementation Details: I followed the existing patterns found in resistance_level and stroke_count:

Model (Record.dart): Added inclination as an @ignore field. It exists only in memory during the workout.

Parsing (TreadmillDeviceDescriptor.dart): Updated the FTMS parser to extract the SInt16 inclination value (Bit 3) instead of skipping it.

Preferences: Created lib/preferences/show_inclination.dart and added a toggle in User Experience Preferences (Default: Off).

UI (Recording.dart): Added logic to conditionally render the inclination value and graph only when the preference is enabled.

What This Approach Does:

✅ Honors existing "display only" philosophy.
✅ Requires zero database schema changes.
✅ No export functionality (avoids data bloat).
✅ Works during the workout only (live visualization).

This is, as of yet, untested! Which is why it's a draft :)

DXCanas committed:
Add @ignore inclination field and FTMS parsing

- Add @ignore inclination field to Record model (transient, not persisted)
- Add inclination metric descriptor to DataHandler
- Add processInclinationFlag method to FitnessMachineDescriptor
- Update TreadmillDeviceDescriptor to parse inclination from FTMS data
- Add getInclination getter method to expose inclination value

Follows existing pattern for display-only metrics like resistance level.

DXCanas committed:
Add inclination statistics tracking

- Add avg/max/min inclination calculation to StatisticsAccumulator
- Update processExportRecord, processDisplayRecord, and processRecord
- Follows same pattern as resistance level statistics
- Only calculates when inclination display is enabled

DXCanas committed:
Add inclination preference toggle and UI display

- Create show_inclination.dart preference file with default disabled
- Add preference toggle to user experience preferences UI
- Add conditional inclination display during workout
- Add inclination chart visualization with terrain icon
- Initialize arrays and controllers for inclination expansion
- Follows exact pattern of resistance level feature

The metric is display-only - not persisted or exported

DXCanas committed:
Fix syntax error in inclination UI

- Remove extra comma after ternary operator closing bracket
- Resolves build error: Expected an identifier at line 2723

DXCanas committed:
Merge remote-tracking branch 'upstream/develop' into add-inclination-display-only

DXCanas committed:
Formatting.

DXCanas committed:
Wrap up inclination support to DisplayRecord and statistics.

Fixing `flutter analyze` errors.

MrCsabaToth commented on lib/devices/device_descriptors/data_handler.dart:
```diff
@@ -160,5 +162,6 @@ abstract class DataHandler {
     caloriesPerMinuteMetric = null;
     resistanceMetric = null;
     strokeCountMetric = null;
+    inclinationMetric = null;
```
I'm thinking if we should also add ramp angle here. We should also note that even though this PR doesn't deal with persistence, so this won't bloat the persisted data size, it can still increase the in-memory footprint of the recorded workout. Hopefully that won't be too bad.

MrCsabaToth commented on lib/devices/device_descriptors/fitness_machine_descriptor.dart:
```diff
@@ -194,6 +194,22 @@ abstract class FitnessMachineDescriptor extends DeviceDescriptor {
+      );
+      byteCounter += 2;
+      // SInt16, Ramp Angle (percentage with 0.1 resolution) - skipped for now
+      byteCounter += 2;
```
If we have ramp angle we'd need to support here.

MrCsabaToth replied:
When it comes to display I expect either the inclination or the ramp angle present. The ramp angle can be displayed in the same row if the inclination is not there. If both present the inclination can take precedence, or maybe it could also be alternating (we have alternation for the statistics display when max and avg are alternating, the alternation and interval are preferences configurable). This would add some complexity, maybe we should do it in a follow-up PR, I fear to bloat this PR.

MrCsabaToth commented on lib/ui/recording/recording_screen.dart:
```diff
@@ -1068,7 +1102,7 @@ class RecordingState extends State<RecordingScreen> {
     ];
 
-    _optionalStatistics = [emptyMeasurement];
+    _optionalStatistics = [emptyMeasurement, emptyMeasurement, emptyMeasurement];
```
Hmmm, why are two blanks added here?

DXCanas replied:
Ah, I think this is holdover from a former iteration where I also tried to add ~Resistance~ another field. But. One at a time.

DXCanas replied:
Actually, nope. Something else at play, having to do with the commit I branched off of. Reason this has taken a minute.

DXCanas replied:
Just to keep you up to date, here's what I'm getting:

The two blanks are needed because of the index alignment approach. Here's why:

Index layout:

`_optionalValues`: needs 3 slots (resistance=0, strokeCount=1, inclination=2)
`_optionalStatistics`: upstream has 1 slot (resistance=0 only)
Since strokeCount doesn't have statistics (it's just a counter, no avg/max), slot 1 in `_optionalStatistics` is unused. But to use the same `_inclinationIndex = 2` for both arrays, we need 3 elements → 2 blank slots added.

MrCsabaToth replied:
Oh, this makes sense! This could be an example when we should comment that in the code.

MrCsabaToth commented on lib/utils/statistics_accumulator.dart:
```diff
@@ -422,6 +485,7 @@ class StatisticsAccumulator {
       maxCadence > 0 ? maxCadence : null,
       maxHeartRate > 0 ? maxHeartRate : null,
       maxResistance > 0 ? maxResistance : null,
+      maxInclinationDisplay > 0 ? maxInclinationDisplay : null,
```
Not sure if we require "Display" postfix for the variable name. The resistance also doesn't have "Display" postfix, and I'm not sure if that is persisted.

MrCsabaToth commented on lib/ui/recording/recording_screen.dart:
```diff
@@ -1106,6 +1142,11 @@ class RecordingState extends State<RecordingScreen> {
+      calculateMinInclination:
+          _showInclination && _onStageStatisticsType != onStageStatisticsTypeNone,
     );
     _zoneIndexes = [null, null, null, null];
```
Inclination won't have zone coloring, just like Resistance level (also display only). I think that's the way to go.

MrCsabaToth reviewed (commented):
Right now exploring the PR. Overall looks good.

DXCanas committed:
Use maxInclination directly in maximumDisplayRecord

Match the pattern used for resistance which uses the internal
variable directly rather than the Display getter.

DXCanas committed:
Add comment explaining optional statistics array indices

Clarify that the 3-slot array matches _optionalValues indices where resistance=0, strokeCount=1 (unused, no stats), and inclination=2.

DXCanas committed:
Refactor inclination display to mirror Resistance's.
Now uses MeasurementRow to display value + avg/max statistic, and RecordingChart for the expandable chart. Both are display-only metrics without zone coloring.

DXCanas commented:
Think I resolved all comments — acknowledging ramp angle, and that it's probably best for a separate PR.

MrCsabaToth reviewed (approved):
I don't want to hold up the PR longer. I'll try to test it on my end in develop once merged
